### ISSUE TITLE
mWeb - Magic link - "Sign in here" leads to Abracadabra page.

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#35782]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.36
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4279569
**Email or phone of affected tester (no customers):** vdargentotest+mweb020224@gmail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. On login page, enter an email you have access to
3. Go to email inbox and copy the magic link
4. Change the link to staging and go to it on an incognito window
5. Tap on "just sign here"

## Expected Result:
The user should be logged in and the LHN should be displayed

## Actual Result:
The user should be logged in and the LHN should be displayed

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/abe31d5a-e192-409c-ba8e-ad4dae50b598

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mountiny** [comment]
might be same rootcause as https://github.com/Expensify/App/issues/35745

**Comment by jcdiprose** [comment]
Hi I'm taking a look at this. An incognito browser window is a different session to the non incognito one so logging in from incognito shouldn't work by design. That being said what is meant by LHN as described in the Expected Result?

**Comment by jcdiprose** [comment]
**Proposal**

`credentials.login` is undefined in component ValidateLoginPage when opening the link in incognito.

To support opening the links in incognito we would need to.

- add a URL parameter to the magic link containing the users email address
- Update Onyx CREDENTIALS login property with the email address
- Check title "Head back to your original tab to continue." to "Head back to your original tab to continue. Or click here" which when clicked would direct to the home page with valid logged in credentials this time

**Comment by jcdiprose** [comment]
@Ollyws my mistake thanks for letting me know

Please re-state the problem that we are trying to solve in this issue.
- Opening magic link from incognito window doesn't complete login session

What is the root cause of that problem?
- Opening the magic link from incognito means there is no active session within Onyx

What changes do you think we should make in order to solve the problem?
- add email address as a URL parameter on the magic link
- on page mount update Onyx state with the email address
- provide the user a link "or click here" that will reload the page taking them to LHN.

What alternative solutions did you explore? (Optional)
- None

**Comment by yuwenmemon** [comment]
@jcdiprose If you look on production you'll see that clicking the URL from the magic link signs the user in on the incognito mode window. How is this related to an active session?

**Comment by jcdiprose** [comment]
@yuwenmemon 
Interesting. I thought it was handled differently. I will take a look at prod soon

**Comment by jcdiprose** [comment]
@yuwenmemon If I understand the reproduction steps correctly it sounds like the start of the process is done from one browser window then the magic link is being opened in a different _incognito_ browser window

**Comment by yuwenmemon** [comment]
@jcdiprose Yes, that's correct. In that case, when you click that "just sign in here" link it should take you right into your account.

**Comment by aswin-s** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

App gets redirected to Abracadabra screen when "Sign in here" is clicked


### What is the root cause of that problem?

This is some what related to https://github.com/Expensify/App/issues/35745

In ValidateLogin page a useEffect is used to redirect user once the code is validated. Looks like we are not waiting for Navigation.ready before calling Navigation.navigate. Also Navigation.navigate() without any parameters doesn't seem to work after navigation refactor.

https://github.com/Expensify/App/blob/5133c29ccf359a22225d18547d6776232bc9aa97/src/pages/ValidateLoginPage/index.website.tsx#L27-L31

### What changes do you think we should make in order to solve the problem?

In `ValidationLoginPage/index.website.tsx` modify below line

https://github.com/Expensify/App/blob/5133c29ccf359a22225d18547d6776232bc9aa97/src/pages/ValidateLoginPage/index.website.tsx#L30

```tsx
Navigation.isNavigationReady().then(() => {
  Navigation.goBack();
});
```
This seems to fix the issue.


https://github.com/Expensify/App/assets/6880914/4a1de9a2-a88f-4f73-8c12-fd4935cb42c6



### What alternative solutions did you explore? (Optional)

None

**Comment by yuwenmemon** [comment]
@aswin-s SGTM, can you whip up a PR ASAP?

**Comment by aswin-s** [comment]
Draft PR is ready https://github.com/Expensify/App/pull/35850

**Comment by zanyrenney** [comment]
@Ollyws @aswin-s you have not accepted the offers, please accept for payout,

**Comment by aswin-s** [comment]
@zanyrenney Accepted the offer
